### PR TITLE
dash/sticky-sidebar

### PR DESCRIPTION
**Pull request by sebastianbochan**
Added a sticky sidebar to the Dashboard.

### CODE REVIEW

**Review by karolkolodziej — CHANGES_REQUESTED** [review]
Looks and works good, but the position is off:
image

**Comment by sebastianbochan** [comment]
@karolkolodziej the issue is in the main branch, that I forked. It's caused by max-width and missing position relative in the demo container.

So it's not a part of this task.

**Review by ddragula — CHANGES_REQUESTED** [review]
Nice! Two comments:

1. When not needed, the scrollbar should not be displayed.
image

2. Do we want such a big margin here?
image

File: ts/Dashboards/EditMode/EditMode.ts [hunk 1]
```diff
@@ -1268,6 +1275,19 @@ namespace EditMode {
...
+     * Sidebar options
+     */
+    export interface SidebarOptions {
+        /**
+         * Whether the sidebar should be sticky when scrolling down or
+         * compontent is out of the view port.
+         *
+         * @default false
+         */
+        sticky?: boolean;
```
Comment (karolkolodziej) [inline ts/Dashboards/EditMode/EditMode.ts:1289]: Please add a vertical demo as an example here.

**Review by karolkolodziej — APPROVED** [review]

**Review by ddragula — APPROVED** [review]
Good job!

**Review by stitot — CHANGES_REQUESTED** [review]
Really nice, and IMO a big improvement that form elements in sidebar are now visible in the viewport when you click charts further down the page, and that they stay visible on scroll! Some further tweaking is needed, though.

If you decrease height of browser window and/or expand multiple accordions in the sidebar the save/cancel button disappears below the fold. That's why I believe the sidebar should consist of 3 parts:

1. Header at top
2. Form with overflow:auto and 100% height
3. Footer with buttons that is absolute positioned at the bottom of sidebar.

Please have a look at the description and illustration at https://app.asana.com/0/1203234853779882/1208043991339536/f that shows this.

**In addition (most likely not stated to clear in dialog/Asana):**
The sidebar is now sticky in relation to `#container`, but I believe we need an option where it's sticky in relation to `body`. I assum it's a bigger refactoring job involved here, so in that case we should perhaps postpone until we start looking at the complete overhaul of editmode.

Otherwise: great job!

**Review by stitot — COMMENTED** [review]
Wow, really like this! 

Perhaps just add something like `box-shadow: rgb(0, 0, 0) 0px 0px 10px 0px` to it so it doesn't bleed into the background?

Nice work!

**Comment by stitot** [comment]
And I see it's sticky in relation to `body` as well? Sweet!

**Comment by stitot** [comment]
I would like background with rounded corners to be on the entire container, so that chart title is inside.

Skjermbilde 2024-10-01 kl  09 44 05

**Review by stitot — COMMENTED** [review]
Nice, but sadly I need the same background/padding on .highcharts-dashboards-component-caption

**Review by stitot — APPROVED** [review]

**Pull request merged**